src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java

```diff
@@ -66,18 +67,19 @@ public FilingApi generateAcspApplicationFiling(
           String acspApplicationId,
           String transactionId,
           String passThroughTokenHeader)
-          throws ServiceException {
+          throws ServiceException, SubmissionNotLinkedToTransactionException {
     var filing = new FilingApi();
     setFilingApiData(filing, acspApplicationId, transactionId, passThroughTokenHeader);
     return filing;
   }
 
   private void setFilingApiData(FilingApi filing, String acspApplicationId, String transactionId,
-                                String passThroughTokenHeader) throws ServiceException {
-    var acspDataDto = acspService.getAcsp(acspApplicationId).orElse(null);
+                                String passThroughTokenHeader) throws ServiceException, SubmissionNotLinkedToTransactionException {
+    var transaction = transactionService.getTransaction(passThroughTokenHeader, transactionId);
+    var acspDataDto = acspService.getAcsp(acspApplicationId, transaction).orElse(null);
     if(acspDataDto != null) {
       var data = new HashMap<String, Object>();
-      var transaction = transactionService.getTransaction(passThroughTokenHeader, transactionId);
```

Comment:
I think the purpose of this being inside if was do this only if DTO is not null

Reply:
We need Transaction to fetch the DTO

Reply:
Sorry missed that. Cool